webapp/app/js/models/job.js

```diff
@@ -55,19 +57,30 @@ treeherder.factory('ThJobModel', [
                 if(_.has(response.data, 'job_property_names')){
                     // the results came as list of fields
                     //we need to convert them to objects
-                    item_list = _.map(response.data.results, function(elem){
-                        var job_obj = _.object(response.data.job_property_names, elem);
-                        return new ThJobModel(job_obj);
-                    });
+                    item_list = [];
+                    var props = response.data.job_property_names;
+                    var propsLen = props.length;
+                    for (var i = 0, resultsLen = response.data.results.length; i < propsLen; i++) {
```

Comment:
Before breaking this stuff out into loops (and making the code much less readable), I wonder if we should consider using lodash (https://lodash.com/) instead of underscore. It's supposed to be quite a bit faster than underscore (http://benmccormick.org/2014/11/12/underscore-vs-lodash/), and indeed uses the technique you're using here to iterate through arrays with map (and other things): https://github.com/lodash/lodash/blob/master/lodash.js#L1443